From dc465a8f672a7bf4e9737d48a68cec1390fd328f Mon Sep 17 00:00:00 2001
From: Antony Abee Prakash XV <antonyabee.prakashxv@adlinktech.com>
Date: Fri, 17 Jul 2020 17:48:30 +0530
Subject: [PATCH 3/3] Add UART support

- This patch will add support for UARTS 1,2,5,4
- Removes conflicting pinctrl_pcie_reg and pcie regulator which
  is not required

Signed-off-by: Antony Abee Prakash XV <antonyabee.prakashxv@adlinktech.com>
---
 arch/arm/boot/dts/lec-imx6.dtsi | 75 ++++++++++++++++++---------------
 1 file changed, 42 insertions(+), 33 deletions(-)

diff --git a/arch/arm/boot/dts/lec-imx6.dtsi b/arch/arm/boot/dts/lec-imx6.dtsi
index 50c86af3a810..6ee8b28653cd 100644
--- a/arch/arm/boot/dts/lec-imx6.dtsi
+++ b/arch/arm/boot/dts/lec-imx6.dtsi
@@ -64,17 +64,6 @@
 		enable-active-high;
 	};
 
-	reg_pcie: regulator-pcie {
-		compatible = "regulator-fixed";
-		pinctrl-names = "default";
-		pinctrl-0 = <&pinctrl_pcie_reg>;
-		regulator-name = "MPCIE_3V3";
-		regulator-min-microvolt = <3300000>;
-		regulator-max-microvolt = <3300000>;
-		gpio = <&gpio3 19 GPIO_ACTIVE_HIGH>;
-		enable-active-high;
-	};
-
 	reg_sensors: regulator-sensors {
 		compatible = "regulator-fixed";
 		pinctrl-names = "default";
@@ -929,12 +918,6 @@
 			>;
 		};
 
-		pinctrl_pcie_reg: pciereggrp {
-			fsl,pins = <
-				MX6QDL_PAD_EIM_D19__GPIO3_IO19	0x1b0b0
-			>;
-		};
-
 		pinctrl_pwm1: pwm1grp {
 			fsl,pins = <
 				MX6QDL_PAD_SD1_DAT3__PWM1_OUT		0x1b0b1
@@ -948,14 +931,30 @@
 		};
 
 		pinctrl_uart1: uart1grp {
-			fsl,pins = <
-				MX6QDL_PAD_CSI0_DAT10__UART1_TX_DATA	0x1b0b1
-				MX6QDL_PAD_CSI0_DAT11__UART1_RX_DATA	0x1b0b1
+                       fsl,pins = <    /* SMARC SER0 */
+				MX6QDL_PAD_SD3_DAT6__UART1_RX_DATA      0x1b0b1
+				MX6QDL_PAD_SD3_DAT7__UART1_TX_DATA      0x1b0b1
+				MX6QDL_PAD_EIM_D19__UART1_CTS_B         0x1b0b1
+				MX6QDL_PAD_EIM_D20__UART1_RTS_B         0x1b0b1
 			>;
 		};
 
-		pinctrl_uart5_1: uart5grp-1 {
-			fsl,pins = <
+		pinctrl_uart2: uart2grp {
+			fsl,pins = <    /* SMARC SER1 */
+				MX6QDL_PAD_EIM_D26__UART2_TX_DATA       0x1b0b1
+				MX6QDL_PAD_EIM_D27__UART2_RX_DATA       0x1b0b1
+			>;
+		};
+
+		pinctrl_uart4: uart4grp {
+			fsl,pins = <    /* SMARC SER3 */
+				MX6QDL_PAD_KEY_COL0__UART4_TX_DATA      0x1b0b1
+				MX6QDL_PAD_KEY_ROW0__UART4_RX_DATA      0x1b0b1
+			>;
+		};
+
+		pinctrl_uart5: uart5grp {
+			fsl,pins = <    /* SMARC SER2 */
 				MX6QDL_PAD_KEY_COL1__UART5_TX_DATA	0x1b0b1
 				MX6QDL_PAD_KEY_ROW1__UART5_RX_DATA	0x1b0b1
 				MX6QDL_PAD_KEY_COL4__UART5_RTS_B	0x1b0b1
@@ -963,15 +962,6 @@
 			>;
 		};
 
-		pinctrl_uart5dte_1: uart5dtegrp-1 {
-			fsl,pins = <
-				MX6QDL_PAD_KEY_ROW1__UART5_TX_DATA	0x1b0b1
-				MX6QDL_PAD_KEY_COL1__UART5_RX_DATA	0x1b0b1
-				MX6QDL_PAD_KEY_ROW4__UART5_RTS_B	0x1b0b1
-				MX6QDL_PAD_KEY_COL4__UART5_CTS_B	0x1b0b1
-			>;
-		};
-
 		pinctrl_usbotg: usbotggrp {
 			fsl,pins = <
 				MX6QDL_PAD_ENET_RX_ER__USB_OTG_ID	0x17059
@@ -1090,7 +1080,6 @@
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_pcie>;
 	reset-gpio = <&gpio7 12 GPIO_ACTIVE_LOW>;
-	vpcie-supply = <&reg_pcie>;
 	epdev_on-supply = <&vgen3_reg>;
 	status = "okay";
 };
@@ -1129,9 +1118,29 @@
 	status = "okay";
 };
 
-&uart1 {
+&uart1 {	/* SMARC SER0 */
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_uart1>;
+	fsl,uart-has-rtscts;
+	status = "okay";
+};
+
+&uart2 {       /* SMARC SER1 */
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart2>;
+	status = "okay";
+};
+
+&uart4 {       /* SMARC SER3 */
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart4>;
+	status = "okay";
+};
+
+&uart5 {       /* SMARC SER2 */
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart5>;
+	fsl,uart-has-rtscts;
 	status = "okay";
 };
 
-- 
2.17.1

